qubesbuilder/config.py

```diff
@@ -611,6 +613,56 @@ def get_stages(self) -> List[str]:
     def get_plugin_manager(self):
         return PluginManager(self.get_plugins_dirs())
 
+    def get_needs(
+        self,
+        component: QubesComponent,
+        dist: QubesDistribution,
+        stage_name: str,
+    ):
+        needs = []
+        stages = component.kwargs.get(dist.distribution, {}).get("stages", [])
+        for stage in stages:
+            if (
+                isinstance(stage, dict)
+                and next(iter(stage)) == stage_name
+                and isinstance(stage[stage_name], dict)
```

Comment:
At least some of this not matching should show a warning or even error. Otherwise it will be hard to find why it doesn't work if you make a typo (like extra `-` making it a list instead of dict).